packages/trpc/server/routers/viewer/bookings.tsx

```diff
@@ -126,7 +144,29 @@ export const bookingsRouter = createProtectedRouter()
         };
 
         const eventManager = new EventManager(ctx.user);
-        // TODO: This is causing duplicate calendar events when updating location
+
+        //delete existing calendar event
+        const credentialsMap = new Map();
+        booking.user?.credentials.forEach((credential) => {
+          credentialsMap.set(credential.type, credential);
+        });
+        const bookingRefsFiltered: BookingReference[] = booking.references.filter(
+          (ref) => !!credentialsMap.get(ref.type)
+        );
+        bookingRefsFiltered.forEach(async (bookingRef) => {
+          if (bookingRef.uid) {
+            if (bookingRef.type.endsWith("_calendar")) {
+              const calendar = getCalendar(credentialsMap.get(bookingRef.type));
+
+              return calendar?.deleteEvent(
+                bookingRef.uid,
+                evt,
+                bookingRef.externalCalendarId
+              );
+            }
+          }
+        });
```

Comment:
Some Qs: Are we returning promises here? Do we want them to run in parallel, on cascade or is it irrelevant?